pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -123,6 +123,12 @@ def processing_fn(results):
             return self._old_device_transform(obj, targs, tkwargs)
         if isinstance(obj, qml.devices.Device):
             return self._device_transform(obj, targs, tkwargs)
+        if obj.__class__.__name__ == "QJIT":
+            raise TransformError(
+                "Functions that are wrapped / decorated with qjit cannot subsequently be"
+                f" transformed with any transform in qml.transforms (attempted {self})."
+                f" For the desired affect, ensure that qjit is applied after {self}."
+            )
```

Comment:
Nice! This works for me. I wonder if it makes sense to do this instead so that it's more clear when custom transforms are used.

```suggestion
        if obj.__class__.__name__ == "QJIT":
            raise TransformError(
                "Functions that are wrapped / decorated with qjit cannot subsequently be"
                f" transformed with a PennyLane transform (attempted {self})."
                f" For the desired affect, ensure that qjit is applied after {self}."
            )
```